src/pages/Upload.tsx

```
@@ -22,6 +22,53 @@
+  // 与后端保持一致的 repo/key 推导（见 server/src/api/index-upload.js）
+  const repoNameFromFile = (name: string) => {
+    const base = (name.replace(/[^A-Za-z0-9._-]/g, '_').replace(/\.[^.]+$/, '') || 'project')
+    const m = base.match(/^(.*?)[@#]([^@#]+)$/)
+    return m ? m[1] : base
+  }
+  const keyFromRepoName = (repo: string) =>
+    ((repo.replace(/[^A-Za-z0-9._-]+/g, '-').replace(/^-+|-+$/g, '')) || 'project').slice(0, 80)
+  const branchFromFile = (name: string) => {
+    const base = (name.replace(/\.[^.]+$/, '') || '')
+    // 1) @ 或 #
+    let m = base.match(/^(.*?)[@#]([^@#]+)$/)
```

ESLint

[error] 'm' is never reassigned. Use 'const' instead.

(prefer-const)